pkg/app/router.go

```diff
@@ -92,6 +93,15 @@ func redirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// To avoid redirect loop
+	if link.URL == link.ID {
+		http.Error(w, "Link id must be different than url", http.StatusBadRequest)
```

Comment:
`Link id` is, IMHO, bit confusing. As a user I would probably confused what this ID is. As far as I understand now `id` is url part provided by the user? If so, I'd recommend changing it to something like `value` or something similar :)

Reply:
Good suggestion! I've renamed it and added some comments to structs